sqlmesh/core/engine_adapter/base.py

```diff
@@ -112,6 +117,12 @@ def is_supported(self) -> bool:
         return self.is_requires_set_catalog or self.is_full_support
 
 
+class CommentCreation(Enum):
+    UNSUPPORTED = 1
+    IN_SCHEMA_DEF = 2
+    COMMENT_COMMAND_ONLY = 3
```

Comment:
Can you move this to `shared.py`? We have `DataObjectType` enum in there which is shared across these adapters. Also can you follow it's implementation of implementing `str, Enum` and then creating the `is_*` properties. I like doing this since it can help avoid doing an import and creating a circular dependency when I just need to check an Enum value is something I would generally expect.